### DISCUSSION TITLE
How to configure DS18B20 as Roomsensor for HK1?

### DISCUSSION THREAD

**Discussion by MarcelFra83** [emsesp/EMS-ESP32#1794] [Q&A]
Hi, I've a Buderus LogamaxPlus (GB182i-20) heater for water and radiators.
The RC310 is in one of the rooms, but I'd like to measure the Roomtemp for hk1 with the DS18B20 sensor connected to ems-ESP Board.
The Sensor was recognized by the ems-ESP and shows the current roomtemp.

Now the Question: How to configure the ems-ESP to use this sensor instead of the RC310 buildin sensor?

Was looking everywhere but could only find solutions with external temp level sent over MQTT, API etc.

Cheers

Marcel

**Comment by proddy** [comment]
Did you play around with https://emsesp.github.io/docs/Special-Functions/#remote-thermostats ?

**Comment by MichaelDvP** [comment]
Uncommon that emsesp is within cable length of a reference room. 
Try with scheduler:timer-mode every few minutes, command `thermostat/hc1/remotetemp`, value: `temperaturesensor/<dallas_id_or_name>`

**Comment by MarcelFra83** [comment]
Hi, thank you for the commands.
I've tried it but it's still not working.
Attached a screenshot of the scheduler entry.

![schedule](https://github.com/emsesp/EMS-ESP32/assets/172313976/ac539e83-ccfe-4df0-9d02-6b1720f33952)

Maybe someone has an idea?

Cheers Marcel

**Reply by proddy** [reply]
can you send your system info and also what you see in the log when the scheduler command is called. This will help us trace down the root cause.

**Comment by MichaelDvP** [comment]
@MarcelFra83 Have you read https://emsesp.github.io/docs/Special-Functions/#remote-thermostats
You also have to tell your thermostat that there is a remote, this can be done on the thermostat or via thermostat/control in ems-esp.
Please rename the temperaturesensor, the id with minus-signs will not work in the dev with formula calculation. 
@proddy should we change the sensor id to underscore, for HA we already change the id.

**Reply by proddy** [reply]
@MichaelDvP yes, we should replace - with _ and also include it in the Web TextBox validation. And then mark this down as a breaking change so people will need to manually rename the temperature sensors. We can also do this in the upgrade() code but it's really not worth the effort.

Do you want to make the change or shall I ?

**Reply by MichaelDvP** [reply]
> Do you want to make the change or shall I ?

Please do it, you have better look what is needed in the test-data. In temperaturesensor it's just change https://github.com/emsesp/EMS-ESP32/blob/a0ca1ce87c9777ea12f5f0a3f49e489ae54a0096/src/temperaturesensor.cpp#L583 and remove https://github.com/emsesp/EMS-ESP32/blob/a0ca1ce87c9777ea12f5f0a3f49e489ae54a0096/src/temperaturesensor.cpp#L561

**Reply by MichaelDvP** [reply]
@proddy I think i have found the test data and made the change in my dev.

**Reply by proddy** [reply]
Great!

**Reply by proddy** [reply]
Have you ever tried standalone/native with the test data? I've never tried on windows, just Linux with WSL

**Comment by MarcelFra83** [comment]
Hi, thank you for the tips but still it's not working.
I've changed the name of the external sensor to "SensorBad" and changed the Scheduler:
![grafik](https://github.com/emsesp/EMS-ESP32/assets/172313976/83b1e194-8e8d-484e-ac9a-165bc57d450e)

Also I've tried to "activate" the remote thermostat, but there is only a hc1 setting in the Device "RC310" so the Main Display.
Under Devices: "GBx72" (I guess this is the Heater) there is no setting with HCx shown.
Now I'm confused. Where do I need to set/activate this remote Thermostat?

Also in the SystemLog with Loglevel "ALL" I can't find the scheduler event...
![grafik](https://github.com/emsesp/EMS-ESP32/assets/172313976/00ad377b-d76f-425e-8fd7-39e6d6402afe)

Any Ideas?

Here are both devices:
[Kessel_GBx72_Trendline_Cerapur_Greenstar Si.csv](https://github.com/user-attachments/files/16116341/Kessel_GBx72_Trendline_Cerapur_Greenstar.Si.csv)
[Thermostat_RC300_RC310_Moduline 3000_1010H_CW400_Sense II_HPC410.csv](https://github.com/user-attachments/files/16116342/Thermostat_RC300_RC310_Moduline.3000_1010H_CW400_Sense.II_HPC410.csv)

Cheers

**Reply by MarcelFra83** [reply]
[log.txt](https://github.com/user-attachments/files/16116352/log.txt)

**Reply by MichaelDvP** [reply]
> Any Ideas?
> 2024-07-06 15:57:09.000 INFO 1: [emsesp] Starting EMS-ESP version 3.6.4

that's really very old. Please update to current release or better try the actual 3.7.0-dev.

**Reply by proddy** [reply]
I see you're running version 3.6.4 which is 8 months old - take the latest 3.7.0 dev version as this release has the features you need

**Reply by proddy** [reply]
this is why we always first ask for the system info @MarcelFra83 - https://github.com/emsesp/EMS-ESP32/discussions/1794#discussioncomment-9952065

**Reply by MarcelFra83** [reply]
:)
So it's updated to 3.7.0_dev22
Now there is something in the log:

2024-07-06 19:11:43.000 INFO 287: [command] Calling command 'thermostat/remotetemp' (Raumtemperatur Remote) with value 23.2 and id 1 on device 0x10
2024-07-06 19:11:43.060 TRACE 288: [emsesp] thermostat(0x38) -W-> thermostat(0x10), RemoteTemp(0x042B), data: 00 E8 09 13 01

When I'm trying to change the hc1 remote with RC100H I get a confirmation "write command sent" but wehen I reload the page it's back on "RC310"
Unter "Devices" there is now an RC100H" with exact the same Temp as the external temp-Sensor.
Still I don't geht how the virtual RC100H is now sending the temperature to HC1?
